test/arguments/escape-no-icu.js

```diff
@@ -0,0 +1,16 @@
+// Mimics Node.js when built without ICU support
+// See https://github.com/sindresorhus/execa/issues/1143
+globalThis.RegExp = class extends RegExp {
+	constructor(regExpString, flags) {
+		if (flags?.includes('u') && regExpString.includes('\\p{')) {
+			throw new Error('Invalid property name');
+		}
+
+		super(regExpString, flags);
+	}
+
+	static isMocked = true;
+};
```

Comment:
I cannot find a simple way to run tests using Node without ICU support. So this mimics it for our purpose.